packages/zowe-explorer/src/commands/ZoweCommandProvider.ts

```diff
@@ -19,18 +19,225 @@ import { Constants } from "../configuration/Constants";
+                    },
+                    {
+                        message: vscode.l10n.t({
+                            message: "Welcome to the integrated terminal for: {0}",
+                            args: [this.terminalName],
+                            comment: ["Terminal Name"],
+                        }),
+                        history: [...this.history.getSearchHistory()].reverse(),
+                        startup: command,
+                        formatCommandLine: (cmd: string) => this.formatCommandLine(cmd, profile),
+                    }
+                );
+                this.terminal = vscode.window.createTerminal({ name: `(${profile.name}) ${this.terminalName}`, pty: this.pseudoTerminal });
+                this.terminal.show();
+            } else {
+                this.outputChannel ??= Gui.createOutputChannel(this.terminalName);
+                this.outputChannel.appendLine(this.formatCommandLine(command, profile));
+                const response = await Gui.withProgress(
+                    {
+                        location: vscode.ProgressLocation.Notification,
+                        title: this.dialogs.commandSubmitted,
+                    },
+                    () => {
+                        return this.runCommand(profile, command);
+                    }
+                );
+                this.outputChannel.appendLine(response);
+                this.outputChannel.show(true);
+                this.history.addSearchHistory(command);
+            }
+        } catch (error) {
+            await AuthUtils.errorHandling(error, { apiType: ZoweExplorerApiType.Command, profile });
+        }
+    }
+
+    public async selectNodeProfile(cmdTree: Definitions.Trees): Promise<imperative.IProfileLoaded> {
+        ZoweLogger.trace("ZoweCommandProvider.selectNodeProfile called.");
+
+        const profileNamesList = ProfileManagement.getRegisteredProfileNameList(cmdTree);
+        if (profileNamesList.length > 0) {
+            const quickPickOptions: vscode.QuickPickOptions = {
+                placeHolder: this.dialogs.selectProfile,
+                ignoreFocusOut: true,
+                canPickMany: false,
+            };
+            const sesName = await Gui.showQuickPick(profileNamesList, quickPickOptions);
+            if (sesName === undefined) {
+                ZoweLogger.info(this.operationCancelled);
+                return;
+            }
+            const profile = this.profileInstance.allProfiles.find((tempProfile) => tempProfile.name === sesName);
+            await this.profileInstance.checkCurrentProfile(profile);
+            if (this.profileInstance.validProfile === Validation.ValidationType.INVALID) {
+                Gui.errorMessage(vscode.l10n.t("Profile is invalid"));
+                return;
+            }
+            return profile;
+        } else {
+            const noProfAvailable = vscode.l10n.t("No profiles available");
+            ZoweLogger.info(noProfAvailable);
```

Comment:
Similar request to the operation cancelled log message - if we want to log a message here, additional context could help surface an issue if no profiles are available.

Reply:
Resolved by 725e369b4